plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreReadAccessOperations.java

```diff
@@ -0,0 +1,340 @@
+                ImmutableMap.of(
+                        GET_ALL_TABLES_WITH_PARAMETER, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, 1,
+                        GET_TABLE, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsWithLikeOverSchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema LIKE 'test%'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT));
+    }
+
+    @Flaky(issue = "For particular query there are 10294+/-1 GET_TABLE requests, therefore it is flaky", match = ".*")
+    @Test
+    public void testSelectColumnsFilterByTableName()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_name = 'test_table_0'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT + 91));
+    }
+
+    @Test
+    public void testSelectColumnsWithLikeOverTableName()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_name LIKE 'test%'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsFilterByColumn()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE column_name = 'name'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsWithLikeOverColumn()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE column_name LIKE 'n%'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT));
+    }
+
+    private void assertMetastoreInvocations(@Language("SQL") String query, Map<Methods, Integer> expectedInvocations)
```

Comment:
Why not use `Multiset` like in `TestHiveMetastoreAccessOperations`?

I think I'd prefer to merge this and `TestHiveMetastoreAccessOperations` (now or later), so keeping things same would be helpful

Reply:
> Why not use Multiset like in TestHiveMetastoreAccessOperations?

It hard to read when I need to provide ~100 elements

Reply:
`addCopies(element, 100)`

Reply:
I miss that one, I changed the assertions like in `TestHiveMetastoreAccessOperations`